pkg/asset/internal/templates.go

```diff
@@ -299,24 +300,31 @@ spec:
         - --allocate-node-cidrs=true
         - --configure-cloud-routes=false
         - --cluster-cidr={{ .PodCIDR }}
-        - --root-ca-file=/etc/kubernetes/secrets/ca.crt
-        - --service-account-private-key-file=/etc/kubernetes/secrets/service-account.key
+        - --root-ca-file=/etc/secrets/ca.crt
+        - --service-account-private-key-file=/etc/secrets/service-account.key
         - --leader-elect=true
-        - --cloud-provider={{ .CloudProvider  }}
+        - --cloud-provider={{ .CloudProvider }}
+        - --kubeconfig=/etc/kubernetes/kubeconfig
```

Comment:
I think it is needed because when removed the tests are failing. Although the service account should be used, it doesn't seem to be. Also, kubeadm mounts the host /etc/kubernetes dir in order to provide the `--kubeconfig` flag. See:

https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/master/manifests.go#L393
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/master/manifests.go#L238-L253
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/master/manifests.go#L96